boards/arm/lpcxpresso55s69/CMakeLists.txt

```diff
@@ -33,4 +33,64 @@ if (CONFIG_BUILD_WITH_TFM)
 	if(${SREC_CAT} STREQUAL SREC_CAT-NOTFOUND)
 		message(FATAL_ERROR "'srec_cat' not found. Please install it, or add it to $PATH.")
 	endif()
+
+	if (CONFIG_TFM_BL2_FALSE)
```

Comment:
I must have missed this on the TFM side, but this seems like a poor name choice? Why not name the config in the positive sense (e.g. `CONFIG_TFM_BL2`) and use logic to get the correct case.

Reply:
This definition was added as part of building with TFM, it's one of the avaialble choices.

This could be something like:
```
choice TFM_BL2
        prompt "BL2"
        default TFM_BL2_CONFIG_FILE_DEFAULT

config TFM_BL2_CONFIG_FILE_DEFAULT
        bool "Use TFM BL2 setting from TFM configuration file"

config TFM_BL2_TRUE
        bool "TFM BL2 enabled"

config TFM_BL2_FALSE
        bool "TFM BL2 disabled"

endchoice
```
so that default is to use the TFM BL2 setting from conf file, or specifically enable/disable this feature.

_Originally posted by @tejlmand in https://github.com/zephyrproject-rtos/zephyr/pull/27780#discussion_r482818453_

Reply:
I wouldn't worry about it, since the poor naming is coming from outside this change.

Reply:
> 

@d3zd3z not

> I must have missed this on the TFM side, but this seems like a poor name choice? Why not name the config in the positive sense (e.g. `CONFIG_TFM_BL2`) and use logic to get the correct case.

@d3zd3z not sure what you mean ?
The reason it is named like this, is because we need a tri-state boolean (not set==Use what is default in TFM, True, False), and as that is not supported in Kconfig, then we need a choice menu.

Please let me know if what you think would be better naming, as we cannot simply use `True/False` here.

Reply:
It just seems bad to have two definitions that must be set opposite with each other. It would be nonsensical to have both `TFM_BL2_TRUE` and `TFM_BL2_FALSE` set. Having 3 flags gives us 8 combination, where we really only want 3. Why not one for using the default, and then one that sets whether BL2 is set when the default isn't set?

Reply:
But, I agree none of them seem to match KConfig very well.

Reply:
> It would be nonsensical to have both TFM_BL2_TRUE and TFM_BL2_FALSE set. Having 3 flags gives us 8 combination

Setting both `TFM_BL2_TRUE=y` and `TFM_BL2_FALSE=y` is not possible because they are under the same choice group under `TFM_BL2`.
This means there are only three options, just like a tristate.
In a choice group, only one of the settings can be active.

Reply:
You can verify for yourself:
```
# Trying to set all three values to `y`
$ cmake -GNinja -DBOARD=v2m_musca_b1_nonsecure -DCONFIG_TFM_BL2_TRUE=y -DCONFIG_TFM_BL2_CONFIG_FILE_DEFAULT=y -DCONFIG_TFM_BL2_FALSE=y  -Bbuild samples/tfm_integration/psa_level_1/
....
$ grep TFM_BL2 build/zephyr/.config
# CONFIG_TFM_BL2_CONFIG_FILE_DEFAULT is not set
CONFIG_TFM_BL2_TRUE=y
# CONFIG_TFM_BL2_FALSE is not set
```

Or use `menuconfig`.

Reply:
I guess my only real complaint is with the names. `xx_TRUE=n` just feels weird (true=no). Maybe something like TFM_USE_BL2, TFM_NO_BL2, or something like that.

Reply:
I see what you mean.
Looking at the `menuconfig`, things looks more logical:
![image](https://user-images.githubusercontent.com/9802583/95423691-ca826700-0941-11eb-8986-ae888e28f858.png)

Maybe using, `TFM_BL2_ENABLE` and `TFM_BL2_DISABLE` would be cleaner:
```
config TFM_BL2_ENABLE
        bool "TFM BL2 enabled"

config TFM_BL2_DISABLE
        bool "TFM BL2 disabled"
```

The `TRUE`, `FALSE`, `DEFAULT` was originally chosen to resemble a tri-state: `True`, `False`, `Unset`.
But a test like:
`if(CONFIG_TFM_BL2_DISABLE)` looks nicer than `if(CONFIG_TFM_BL2_FALSE)`